File: Java/src/gestorAplicacion/gestorMusica/Musica.java
package gestorAplicacion.gestorMusica;

import java.io.Serializable;

public abstract class Musica implements Serializable{

	private static final long serialVersionUID = 1L;
	protected String nombre;
    protected int reproducciones;

    protected Musica(String nombre){
        this.nombre = nombre;
        this.reproducciones = 0;
    }

    public String getNombre() {
		return nombre;
	}

	public void setNombre(String nombre) {
		this.nombre = nombre;
	}

    public int getReproducciones() {
		return reproducciones;
	}

	public void setReproducciones(int reproducciones) {
		this.reproducciones = reproducciones;
	}
     
    public abstract void aumentarReproducciones();

    @Override
    public abstract String toString();

}

File: Python/src/gestorAplicacion/gestorMusica/musica.py
''' Módulo musica.py
    Autores: Carolina Álvarez Murillo, Miller Johan Chica Acero,
             Tomás Rodríguez Taborda, Jerónimo Ledesma Patiño,
             Catalina Restrepo Salgado
    Este módulo contiene la clase Música, de quien van a heredar Canción, Lista y MeGusta
'''

# Esta clase describe los principales atributos y comportamientos de la música que se
# va a gestionar en la aplicación
class Musica: 

    def __init__(self, nombre):
        self.__nombre = nombre
        self.__reproducciones = 0
    
    def getNombre(self):
        return self.__nombre

    def setNombre(self, nombre):
        self.__nombre = nombre
    
    def getReproducciones(self):
        return self.__reproducciones
    
    def setReproducciones(self, reproducciones):
        self.__reproducciones = reproducciones

    def aumentarReproducciones():
        pass

    def __str__():
        pass